### PR TITLE
Added support for select tag

### PR DESCRIPTION
**Pull request by abhisheksavant**
1. Added support for select tag
2. Updated _triggerEvents method to handle focus & blur events

@alexindigo @jskulski

### CODE REVIEW

**Comment by jskulski** [comment]
conflicted

**Comment by abhisheksavant** [comment]
Resolved conflict

File: mixin.js [hunk 1]
```diff
@@ -351,12 +370,21 @@
...
         for (i=0; i<event.length; i++)
         {
-          // detect key- events from touch- events
-          // TODO: Add support for mouse- events
-          eventObject = this[event[i].match(/^key/) ? '_createTypeEvent' : '_createTouchEvent'].apply(this, [event[i]].concat(extras));
-
-          // trigger event on target
-          target.trigger(eventObject);
+          switch(event) {
+            case 'focus':
```
Comment (jskulski) [inline mixin.js:374]: is there a way we can begin to make this stronger than magic strings? @alexindigo

File: mixin.js [hunk 2]
```diff
@@ -217,6 +217,25 @@
...
+
+        // get the first element of the selection
+        target = (typeof target == 'string' ? this.$(target) : target).first();
+
+        // sync all event handlers
+        setTimeout(function() {
+
+          this._triggerEvents(target, ['focus', 'touchstart', 'touchend', ['mousedown', 'mouseup', 'mouseclick', 'click']], target, function() {
+            target.find('option[value=' + optionValue + ']').attr('selected', true).trigger('change');
+            this._triggerEvents(target, ['blur'], target, callback);
```
Comment (jskulski) [inline mixin.js:231]: for example, this is used here too and you have to know to wrap it as an array [ ]

File: mixin.js (hunk 1 above)
Comment (alexindigo) [inline mixin.js:374]: Yes, I have an idea. Going to work on it next week.

Should simplify usage of `_triggerEvents` and make it easier to extend.

Comment (alexindigo) [inline mixin.js:374]: Or you're talking about replacing strings with "constants"?

File: mixin.js (hunk 2 above)
Comment (alexindigo) [inline mixin.js:231]: It's not really necessary here. But yes, more docs on internal API is due.

**Comment by alexindigo** [comment]
Thank you for PR @abhisheksavant, can you update README file as well?

**Comment by abhisheksavant** [comment]
Updated README

File: README.md [hunk 3]
```diff
@@ -125,6 +125,23 @@ Simulates user's typing, with all related events and proper timing.
 }
 ```
 
+### .select(selector|target, optionValue, [callback])
+
+Simulates user's select drop-down action, with all related events and proper timing.
+
+```javascript
+'Set max value for select drop-down': function(done)
```
Comment (alexindigo) [inline README.md]: `max`?

File: README.md [hunk 4]
```diff
@@ -125,6 +125,23 @@ Simulates user's typing, with all related events and proper timing.
...
+{
+  this.type('[data-role=selectTag]', '1000', function()
```
Comment (alexindigo) [inline README.md]: `this.type`?

Comment (abhisheksavant) [inline README.md]: I changed this stuff ... my gitx is super messed up sorry about that.

**Comment by abhisheksavant** [comment]
Updated PR.

**Comment by alexindigo** [comment]
We have two of Abhishek here :) http://ia.gs/Zjd3

**Pull request merged**